river/expert/bandit.py

```diff
@@ -0,0 +1,269 @@
+import abc
+import copy
+import typing
+
+import numpy as np
```

Comment:
In general, we prefer to use data types that are native to Python. For instance, we've found that the `random` module is faster than `np.random` for small sample sizes. Also numpy arrays are slower than Python lists for most small tasks. I don't mind numpy per se, but I would only use it for linear algebra, not for storing variables. The philosophy of River is to stay as close as possible to the Python standard library :)

Reply:
Ok ! 

If necessary I can get rid of the numpy dependency completely but I'll have to implement argmax (which is not part of the Python stdlib).

Reply:
If you're just doing this in one place you can do:

```py
def argmax(l):
    return max(range(len(l)), key=l.__getitem__)
```

Either make this a function or put inline and add a comment :)

Reply:
Ok thank you !

Also `math.log`, `math.sqrt` etc are not vectorized by default. Should I use listcomp instead ?

Reply:
Actually I think `argmax` should be put in `river.util.math` ; what do you think ?

Reply:
> Also `math.log`, `math.sqrt` etc are not vectorized by default. Should I use listcomp instead?

Yes a list comprehension you should do fine. You can quickly test by yourself the performance versus numpy.

> Actually I think argmax should be put in river.util.math; what do you think?

Agreed!

Reply:
done